changelog/exceptions-opt.dd

```diff
@@ -0,0 +1,26 @@
+Exception trapping can now be disabled via `--DRT-exopt=trap:0`
+
+Previously it was only possible to disable the trapping of exception by setting
+the global variable `rt_trapExceptions` to `false`.
+For example, you can now immediately open `gdc` at the uncaught exception:
+
+$(CONSOLE
+> gdb -ex run --args <my-program> "--DRT-exopt=trap:0"
+object.Exception@/home/seb/dlang/druntime/test/exceptions/src/rt_opt.d(3): foo
+----------------
+Program received signal SIGILL, Illegal instruction.
+0x00005555555a65e9 in _D4core8demangle__T8DemangleTSQBcQBa7NoHooksZQBa__T10doDemangleSQCkQCi__TQCcTQBwZQCk16parseMangledNameZQCaMFNaNbNfZAa ()
+(gdb) bt
+#0  0x00005555555a65e9 in _D4core8demangle__T8DemangleTSQBcQBa7NoHooksZQBa__T10doDemangleSQCkQCi__TQCcTQBwZQCk16parseMangledNameZQCaMFNaNbNfZAa ()
+#1  0x000055555559b194 in rt.backtrace.dwarf.traceHandlerOpApplyImpl(const(void*[]), scope int(ref ulong, ref const(char[])) delegate) ()
+#2  0x0000555555590349 in core.runtime.defaultTraceHandler(void*).DefaultTraceInfo.opApply(scope int(ref ulong, ref const(char[])) delegate) const ()
+#3  0x00005555555902e2 in core.runtime.defaultTraceHandler(void*).DefaultTraceInfo.opApply(scope int(ref const(char[])) delegate) const ()
+#4  0x000055555558797c in object.Throwable.toString(scope void(const(char[])) delegate) const ()
+#5  0x0000555555589000 in rt.dmain2.formatThrowable(object.Throwable, scope void(const(char[])) nothrow delegate) ()
+#6  0x0000555555585650 in _d_print_throwable ()
+#7  0x00005555555857e9 in _d_throwdwarf ()
+#8  0x00005555555850bc in D main (args=...) at rt_opt.d:3
```

Comment:
Something doesn't feel right with this stack trace :thinking: . This actually looks like a bug in runtime - I don't think this signal should be raised while trying to print the exception, the program should actually raise `SIGABRT`: https://github.com/dlang/druntime/blob/master/src/rt/dwarfeh.d#L232

Reply:
I opened an issue for this, s.t. it's not forgotten: https://issues.dlang.org/show_bug.cgi?id=18226 and we can investigate this separately.

Reply:
I'd say it would be better to fix the bug than release a buggy feature.